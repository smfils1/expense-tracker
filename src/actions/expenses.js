import uuid from 'uuid'


//Actions Generators
export const addExpense = ({
    description = '',
    amount = 0,
    createdAt = 0,
    note = ''
}={}) => ({
    type: 'ADD_EXPENSE',   
    expense:{
        id: uuid(),
        description,
        amount,
        createdAt,
        note
    }
})

export const removeExpense = (id) => ({
    type: 'REMOVE_EXPENSE',  
    id 
})

export const editExpense = (id,updates = {}) => ({
    type: 'EDIT_EXPENSE',  
    id,
    updates
})
