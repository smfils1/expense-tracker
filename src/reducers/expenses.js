//Default States
const expensesDefaultState = [];

//Reducers
export default (state = expensesDefaultState, action)=>{
    switch(action.type){
        case 'ADD_EXPENSE':
            return [...state, action.expense]
        case 'EDIT_EXPENSE':
            return state.map(expense => {
                if (expense.id === action.id){
                    return {...expense,...action.updates}
                }
                return expense
            })
        case 'REMOVE_EXPENSE':
            return state.filter(expense => expense.id !== action.id)
            
        default:
            return state
    }
}