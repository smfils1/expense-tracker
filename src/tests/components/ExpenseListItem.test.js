import React from 'react'
import {shallow} from 'enzyme'
import ExpenseListItem from "../../components/ExpenseListItem"
import expenses from "../fixtures/expenses"

describe('<ExpenseListItem/>',()=>{
    test('Should render an expense list correctly',()=>{
        const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>)
        expect(wrapper).toMatchSnapshot()
    })
})