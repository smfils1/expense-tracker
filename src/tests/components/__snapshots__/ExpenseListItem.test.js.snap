// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseListItem/> Should render an expense list correctly 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      d1
    </h3>
  </Link>
  <p>
    0
     - 
    432000000
  </p>
</div>
`;
